pkg/resiliency/resiliency.go

```diff
@@ -1,132 +1,508 @@
+		r.circuitBreakers[name] = &cb
+	}
+
+	for name, t := range buildingBlocks.Services {
+		r.services[name] = PolicyNames{
+			Timeout:        t.Timeout,
+			Retry:          t.Retry,
+			CircuitBreaker: t.CircuitBreaker,
+		}
+		r.serviceCBs[name] = &circuitBreakerInstances{
+			cbs: make(map[string]*breaker.CircuitBreaker, 10),
+		}
+	}
+
+	for name, t := range buildingBlocks.Actors {
+		scope, err := ParseActorCircuitBreakerScope(t.CircuitBreakerScope)
+		if err != nil {
+			return err
+		}
+		r.actors[name] = ActorPolicyNames{
+			PolicyNames: PolicyNames{
+				Timeout:        t.Timeout,
+				Retry:          t.Retry,
+				CircuitBreaker: t.CircuitBreaker,
+			},
+			CircuitBreakerScope: scope,
+		}
+		r.actorCBCaches[name], err = lru.New(t.CircuitBreakerCacheSize)
+		if err != nil {
+			return err
+		}
+	}
+
+	for name, t := range buildingBlocks.Components {
+		r.components[name] = PolicyNames{
+			Timeout:        t.Timeout,
+			Retry:          t.Retry,
+			CircuitBreaker: t.CircuitBreaker,
+		}
+	}
+
+	for name, t := range buildingBlocks.Routes {
+		name = strings.TrimPrefix(name, "/") // Ignore slash prefix.
+		r.routes[name] = PolicyNames{
+			Timeout:        t.Timeout,
+			Retry:          t.Retry,
+			CircuitBreaker: t.CircuitBreaker,
+		}
+	}
+
+	return nil
+}
+
+// RoutePolicy returns the policy for a route.
+func (r *Resiliency) RoutePolicy(ctx context.Context, name string) Runner {
+	var t time.Duration
+	var rc *retry.Config
+	var cb *breaker.CircuitBreaker
+	name = strings.TrimPrefix(name, "/") // Ignore slash prefix.
+	operationName := fmt.Sprintf("route[%s]", name)
+	if r == nil {
+		return Policy(ctx, operationName, t, rc, cb)
+	}
+	policyNames, ok := r.routes[name]
+	if ok {
+		if policyNames.Timeout != "" {
+			t = r.timeouts[policyNames.Timeout]
+		}
+		if policyNames.Retry != "" {
+			rc = r.retries[policyNames.Retry]
+		}
+		if policyNames.CircuitBreaker != "" {
+			cb = r.circuitBreakers[policyNames.CircuitBreaker]
+		}
+	}
+
+	return Policy(ctx, operationName, t, rc, cb)
+}
+
+// RoutePolicy returns the policy for a service endpoint.
+func (r *Resiliency) EndpointPolicy(ctx context.Context, service string, endpoint string) Runner {
+	var t time.Duration
+	var rc *retry.Config
+	var cb *breaker.CircuitBreaker
+	operationName := fmt.Sprintf("endpoint[%s, %s]", service, endpoint)
+	if r == nil {
+		return Policy(ctx, operationName, t, rc, cb)
+	}
+	policyNames, ok := r.services[service]
+	if ok {
+		if policyNames.Timeout != "" {
+			t = r.timeouts[policyNames.Timeout]
+		}
+		if policyNames.Retry != "" {
+			rc = r.retries[policyNames.Retry]
+		}
+		if policyNames.CircuitBreaker != "" {
+			template := r.circuitBreakers[policyNames.CircuitBreaker]
+			ep, ok := r.serviceCBs[service]
+			if ok {
+				cb = ep.Get(endpoint, template)
+			}
+		}
+	}
+
+	return Policy(ctx, operationName, t, rc, cb)
+}
+
+// RemoveEndpoint removes an endpoint from a service.
+// This would be used when a service is taken out of service.
```

Comment:
For my own edification, this is just saying that a pod hosting something has gone down so we don't need to track it anymore, right?

Reply:
Correct